### ISSUE TITLE
🐛 [firebase_core] Error: Member not found: 'FirebaseAppPlatform.verify'

### ISSUE THREAD

**Issue by wayne-wang-1119** [firebase/flutterfire#12194]
## Bug report

**Describe the bug**
I used prev recommended solutions to try run my app, but everytime I run `flutter run` it keeps crashing. The error I get is always:
```
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/firebase_core-2.24.2/lib/src/firebase_app.dart:18:25: Error: Member not found: 'FirebaseAppPlatform.verify'.        
    FirebaseAppPlatform.verify(_delegate);
                        ^^^^^^
Waiting for connection from debug service on Chrome...             16.0s
Failed to compile application.
```

### Steps to reproduce

1. use the same pub as me? 
```
  firebase_core_platform_interface:
  firebase_core: 2.24.2
  firebase_auth: 4.16.0
  cloud_firestore: 4.14.0
```
2. added the constraint here, which is supposed to be a fix:
```
dependency_overrides:
  firebase_core_platform_interface: 4.5.1
```

### Expected behavior

To be able to compile.

### Sample project

Providing a minimal example project which demonstrates the bug in isolation from your main App _greatly_ enhances the chance of a timely fix.
Please link to the public repository URL.

---

## Additional context
here is the output of compacted packages
```
Dart SDK 3.1.5
Flutter SDK 3.13.9
mobile 1.0.0+1

dependencies:
- cloud_firestore 4.14.0 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.6
- firebase_auth 4.16.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.24.2 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 4.5.1 [collection flutter flutter_test meta plugin_platform_interface]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math web sky_engine]
- flutter_signin_button 2.0.0 [flutter font_awesome_flutter]
- font_awesome_flutter 9.2.0 [flutter]
- google_sign_in 6.1.6 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- http 1.1.0 [async http_parser meta]
- just_audio 0.9.36 [just_audio_platform_interface just_audio_web audio_session rxdart path path_provider async uuid crypto meta flutter]
- provider 6.1.1 [collection flutter nested]

dev dependencies:
- flutter_lints 3.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection material_color_utilities meta source_span stream_channel string_scanner term_glyph web]

dependency overrides:
- firebase_core_platform_interface 4.5.1 [collection flutter flutter_test meta plugin_platform_interface]

transitive dependencies:
- _fe_analyzer_shared 61.0.0 [meta]
- _flutterfire_internals 1.3.16 [collection firebase_core firebase_core_platform_interface flutter meta]
- analyzer 5.13.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- args 2.4.2
- async 2.11.0 [collection meta]
- audio_session 0.1.18 [flutter flutter_web_plugins rxdart meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- cloud_firestore_platform_interface 6.1.0 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.9.0 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]      
- collection 1.17.2
- convert 3.1.1 [typed_data]
- crypto 3.0.3 [typed_data]
- fake_async 1.3.1 [clock collection]
- ffi 2.1.0
- file 7.0.0 [meta path]
- firebase_auth_platform_interface 7.0.9 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.8.13 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core_web 2.10.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- glob 2.1.2 [async collection file path string_scanner]
- google_identity_services_web 0.2.2 [js meta]
- google_sign_in_android 6.1.21 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.6.5 [flutter google_sign_in_platform_interface pigeon]
- google_sign_in_platform_interface 2.4.5 [flutter plugin_platform_interface]
- google_sign_in_web 0.12.2+1 [flutter flutter_web_plugins google_identity_services_web google_sign_in_platform_interface http js]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- js 0.6.7 [meta]
- just_audio_platform_interface 4.2.2 [flutter plugin_platform_interface]
- just_audio_web 0.4.9 [just_audio_platform_interface flutter flutter_web_plugins]
- lints 3.0.0
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.9.1
- nested 1.0.0 [flutter]
- package_config 2.1.0 [path]
- path 1.8.3
- path_provider 2.1.2 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- path_provider_android 2.2.2 [flutter path_provider_platform_interface]
- path_provider_foundation 2.3.2 [flutter path_provider_platform_interface]
- path_provider_linux 2.2.1 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.1.2 [flutter platform plugin_platform_interface]
- path_provider_windows 2.2.1 [ffi flutter path path_provider_platform_interface win32]
- pigeon 11.0.1 [analyzer args collection meta path yaml]
- platform 3.1.4
- plugin_platform_interface 2.1.8 [meta]
- pub_semver 2.1.4 [collection meta]
- rxdart 0.27.7
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- sprintf 7.0.0
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.0 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- typed_data 1.3.2 [collection]
- uuid 4.2.2 [crypto sprintf meta]
- vector_math 2.1.4
- watcher 1.1.0 [async path]
- web 0.1.4-beta
- win32 5.1.1 [ffi]
- xdg_directories 1.0.4 [meta path]
- yaml 3.1.2 [collection source_span string_scanner]
```

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
PS C:\Users\wangh\Desktop\R\mobile> flutter doctor -v
[√] Flutter (Channel stable, 3.13.9, on Microsoft Windows [Version 10.0.22621.3007], locale en-US)
    • Flutter version 3.13.9 on channel stable at C:\Users\wangh\Desktop\SideProject\findmoney\messenger_greet_and_meet\lib\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (3 months ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2022 17.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    • Visual Studio Build Tools 2022 version 17.8.34330.188
    • Windows 10 SDK version 10.0.22621.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.85.2)
    • VS Code at C:\Users\wangh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.80.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.3007]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 120.0.6099.225
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 120.0.2210.144

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

**Comment by danagbemava-nc** [comment]
Hi @wayne-wang-1119, you don't manually need to add he `firebase_core_platform_interface` to your project. Can you remove it to see if you're able to build your app?

And is there any reason why you added it in the first place?

**Comment by wayne-wang-1119** [comment]
Wow that did it! Thanks! 

I included it because in other issues addressing this problem the common fix is to have it set to 4.5.1. I guess the bug is fixed.

**Comment by danagbemava-nc** [comment]
Great. 

I'll be closing this as resolved.
